### PR TITLE
add named args to check-manifest.sh

### PR DESCRIPTION
**Pull request by neha170**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

check-manifests.sh: replace positional args with named, populate values by default

###### Change Log  

- check-manifests.sh: replace positional args with named
- check-manifests.sh: populate values by default
- this will make it easier for user to call check-manifests.sh script directly, instead of relying on the make target
- call the script directly in check-manifests.yml
- we can remove the now defunct make target, making our makefile leaner

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Locally: successfully invoked check-manifests.sh directly
- Locally: successfully invoked modified check-manifests make target
- Pipeline: pipeline successfully passed when no toolchain specs were modified, and successfully caught error when toolchain spec was modified, without supporting change in manifest file

### CODE REVIEW

File: SPECS/sed/sed.spec [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Summary:        Stream editor
 Name:           sed
 Version:        4.9
-Release:        1%{?dist}
+Release:        2%{?dist}
```
Comment (neha170) [inline SPECS/sed/sed.spec:4]: This is a test change to see if check-manifests pipeline (PR check) is working correctly

File: .github/workflows/check-manifests.yml [hunk 2]
```diff
@@ -26,12 +26,12 @@ jobs:
     - name: Check x86_64 manifests
       run: |
         pushd toolkit
-        make check-x86_64-manifests
+        ./scripts/toolchain/check_manifests.sh -a "x86_64"
```
Comment (dmcilvaney) [inline .github/workflows/check-manifests.yml:29]: I feel like we want to have the pipelines use the tools the same way a user does. Is there a reason we want to change the users' behavior here?

**Review by dmcilvaney — APPROVED** [review]

File: .github/workflows/check-manifests.yml [hunk 3]
```diff
@@ -26,12 +26,12 @@ jobs:
...
         popd
 
     - name: Check aarch64 manifests
       run: |
         pushd toolkit
-        make check-aarch64-manifests
+        ./scripts/toolchain/check_manifests.sh -a "aarch64"
         popd
       if: always()
```
Comment (dmcilvaney) [inline .github/workflows/check-manifests.yml:37]: What I have been trying to do is add a corrective action for the user into the pipeline.
Something like:
```bash
echo ##########
echo "Ensure toolchain and pkggen manifests (./toolkit/resources/manifests/package/*) match the versions in the .spec files"
echo "Run 'make check-manifests' to validate locally"
echo ##########
```

File: toolkit/scripts/toolchain/check_manifests.sh [hunk 4]
```diff
@@ -4,12 +4,52 @@
...
+DISTRO_MACRO=$(make --no-print-directory -s printvar-DIST_VERSION_MACRO  2> /dev/null)
+
+if [ "$#" -eq 0 ]
+then
+    echo -e "ERROR: Missing required arg\n\n"
+    help >&2
+    exit 1
+fi
+
+while (( "$#")); do
```
Comment (cwize1) [inline toolkit/scripts/toolchain/check_manifests.sh:33]: nit: Use `getopts` instead  of processing it manually.

File: toolkit/scripts/toolchain/check_manifests.sh [hunk 5]
```diff
@@ -4,12 +4,52 @@
...
+    echo "[OPTIONAL] -t -> space-separated list of toolchain specs"
+    echo "[OPTIONAL] -s -> SPECS_DIR"
+    echo "[OPTIONAL] -m -> TOOLCHAIN_MANIFESTS_DIR"
+    echo "[OPTIONAL] -d -> DIST_TAG"
+    echo "[OPTIONAL] -i -> DISTRO_MACRO"
+    echo "[OPTIONAL] -h -> print this help dialogue and exit"
+}
+
+# assign default value
+TOOLCHAIN_SPEC_LIST=$(make --no-print-directory -s printvar-toolchain_spec_list  2> /dev/null)
```
Comment (cwize1) [inline toolkit/scripts/toolchain/check_manifests.sh:20]: So, the `Makefile` is dependent on this script and this script is dependent on the `Makefile`? I can't say I'm particularly a fan of interdependencies like this, even if it does "work".

Would it make sense to add variants of `check-aarch64-manifests` and `check-x86_64-manifests` that don't include dependency lists?

File: toolkit/scripts/toolchain.mk [hunk 6]
```diff
@@ -97,23 +97,9 @@ copy-toolchain-rpms:
 # check that the manifest files only contain RPMs that could have been generated from toolchain specs.
 check-manifests: check-x86_64-manifests check-aarch64-manifests
 check-aarch64-manifests: $(toolchain_files)
-	cd $(SCRIPTS_DIR)/toolchain && \
-		./check_manifests.sh \
-			"$(toolchain_spec_list)" \
-			"$(SPECS_DIR)" \
```
Comment (cwize1) [inline toolkit/scripts/toolchain.mk:103]: What if any of these values are overridden by the user? I think `SPECS_DIR` is commonly overridden value by product teams.

Comment (neha170) [reply toolkit/scripts/toolchain.mk:103]: It has the optional parameter "-s" for overriding default value of SPECS_DIR

Comment (cwize1) [reply toolkit/scripts/toolchain.mk:103]: Yes, but the `Makefile` isn't passing it down.

File: .github/workflows/check-manifests.yml (hunk 2 above)
Comment (neha170) [reply .github/workflows/check-manifests.yml:29]: I think it's a great step in empowering the user to call scripts/ tools directly without having to invoke make. Would help us redesign/ get rid of make in the future more easily.

**Pull request merged**